src/main/webapp/app/entities/course/course.service.ts

```diff
@@ -310,6 +311,176 @@ export class CourseScoresService {
     }
 }
 
+@Injectable()
+export class CourseScoreCalculationService {
+
+    private SCORE_NORMALIZATION_VALUE = 0.01;
+    private courses: Course[] = [];
+
+    constructor(private dateUtils: JhiDateUtils, private http: HttpClient) { }
+
+    calculateTotalScores(courseExercises: Exercise[]): Map<string, number> {
+        const scores = new Map<string, number>();
+        let absoluteScore = 0.0;
+        let maxScore = 0;
+        courseExercises.forEach( exercise => {
+          if (exercise.maxScore !== null) {
+              maxScore = maxScore + exercise.maxScore;
+              const participation: Participation = this.getParticipationForExercise(exercise);
+              if (participation !== undefined) {
+                  const result: Result = this.getResultForParticipation(participation, exercise.dueDate);
+                  absoluteScore = absoluteScore + result.score;
+              }
+          }
+        });
+        scores.set('absoluteScore', absoluteScore);
+        if (maxScore > 0) {
+            scores.set('relativeScore', this.round((absoluteScore / maxScore) * 100, 2));
+        } else {
+            scores.set('relativeScore', 0);
+        }
+        scores.set('maxScore', maxScore);
+        return scores;
+    }
+
+    private round(value, exp) { // helper function to make actually rounding possible
```

Comment:
Somehow the rounding function doesn't properly work...

Reply:
can you tell me what is missing my local tests work.

Reply:
It was a problem related to the calculations that I've mentioned previously in my comments, the method  works. I just changed the number of  decimal point to 1, as I discussed it with Stephan.

Reply:
Everything is fine now, you don't need to change a thing :)